plasmapy/diagnostics/tests/test_proton_radiography.py

```diff
@@ -0,0 +1,340 @@
+
+    # If an array was provided to the constructor, reduce to a single
+    # length scale now.
+    if L.size > 1:
+        L = np.max(L)
+
+    if name == "empty":
+        pass
+
+    elif name == "constant_bz":
+        Bz = np.ones(grid.shape) * 10 * u.T
+        grid.add_quantities(B_z=Bz)
+
+    elif name == "constant_ex":
+        Ex = np.ones(grid.shape) * 5e8 * u.V / u.m
+        grid.add_quantities(E_x=Ex)
+
+    elif name == "axially_magnetized_cylinder":
+        a = L / 4
+        radius = np.linalg.norm(grid.grid[..., 0:2] * grid.unit, axis=3)
+        Bz = np.where(radius < a, 100 * u.T, 0 * u.T)
+        grid.add_quantities(B_z=Bz)
+
+    elif name == "electrostatic_discontinuity":
+        a = L / 2
+        delta = a / 120
+
+        radius = np.linalg.norm(grid.grid[..., 0:2] * grid.unit, axis=3)
+        z = grid.grids[2]
+
+        potential = (1 - erf(z / delta)) * np.exp(-((radius / a) ** 2)) * u.V
+
+        Ex, Ey, Ez = np.gradient(potential, grid.dax0, grid.dax1, grid.dax2)
+
+        grid.add_quantities(E_x=Ex, E_y=Ey, E_z=Ez, phi=potential)
+
+    elif name == "electrostatic_gaussian_sphere":
+        a = L / 3
+        b = L / 2
+        radius = np.linalg.norm(grid.grid, axis=3)
+        arg = (radius / a).to(u.dimensionless_unscaled)
+        potential = np.exp(-(arg ** 2)) * u.V
+
+        Ex, Ey, Ez = np.gradient(potential, grid.dax0, grid.dax1, grid.dax2)
+
+        Ex = -np.where(radius < b, Ex, 0)
+        Ey = -np.where(radius < b, Ey, 0)
+        Ez = -np.where(radius < b, Ez, 0)
+
+        grid.add_quantities(E_x=Ex, E_y=Ey, E_z=Ez, phi=potential)
+
+    else:
+        raise ValueError(
+            "No example corresponding to the provided name " f"({name}) exists."
+        )
+
+    # If any of the following quantities are missing, add them as empty arrays
+    req_quantities = ["E_x", "E_y", "E_z", "B_x", "B_y", "B_z"]
+
+    for q in req_quantities:
+        if q not in list(grid.ds.data_vars):
+            unit = grid.recognized_quantities[q].unit
+            arg = {q: np.zeros(grid.shape) * unit}
+            grid.add_quantities(**arg)
+
+    return grid
+
+
+def run_1D_example(name):
+    """
+    Run a simulation through an example with parameters optimized to
+    sum up to a lineout along x. The goal is to run a realtively fast
+    sim with a quasi-1D field grid that can then be summed to get good
+    enough statistics to use as a test.
+    """
+    grid = _test_grid(name, L=1 * u.mm, num=50)
+
+    # Cartesian
+    source = (0 * u.mm, -10 * u.mm, 0 * u.mm)
+    detector = (0 * u.mm, 200 * u.mm, 0 * u.mm)
+
+    # Catch warnings (test fields aren't well behaved at edges)
+
+    sim = prad.SyntheticProtonRadiograph(grid, source, detector, verbose=False)
+    sim.create_particles(1e4, 3 * u.MeV, max_theta=0.1 * u.deg)
+    sim.run()
+
+    size = np.array([[-1, 1], [-1, 1]]) * 10 * u.cm
+    bins = [200, 60]
+    hax, vax, values = sim.synthetic_radiograph(size=size, bins=bins)
```

Comment:
this implementation is officially a hack now :stuck_out_tongue_winking_eye:

Reply:
Does this satisfy the 1 pun/1000 lines of code requirement, or do I need a more obvious one? :P